database/insertData/dev/03_template_A_featureShowcase.js

```diff
@@ -1293,6 +1293,121 @@ exports.queries = [
             create: [
               ${coreActions}
               ${devActions}
+              {
+                actionCode: "scheduleAction"
+                trigger: ON_APPLICATION_CREATE
+                sequence: 100
+                parameterQueries:{
+                  eventCode: "warn1"
+                  duration: { minute: 2 }
+                }
+              }
+              {
+                actionCode: "scheduleAction"
+                sequence: 102
+                trigger: ON_APPLICATION_CREATE
+                parameterQueries:{
+                  eventCode: "exp1"
+                  duration: { minute: 4 }
+                }
+              }
```

Comment:
I have tried something just to understand how this would work - I think we need to decide/add prevention to that.

Steps:
- started a Feature showcase application
- waited to get first warning
- continued with application - while I was working on it my application was set to EXPIRED
- clicked on "Review and summary" and then Submitted
- The application has done through

I'm unsure if we should:
- Look the application at this point - disabling the "Review and Submit"
- Don't expire application if the section is alive - probably more tricky to do
- Allow going until the resume page, but show problem to submit - Application is expired

Screenshots:
![Screen Shot 2021-08-12 at 3 32 11 PM](https://user-images.githubusercontent.com/16461988/129134709-56c9be27-f043-473c-a9c0-b00453df9d2b.png)
![Screen Shot 2021-08-12 at 3 31 41 PM](https://user-images.githubusercontent.com/16461988/129134716-a44ea37e-aee7-4dc4-b480-a48076f963bf.png)
![Screen Shot 2021-08-12 at 3 31 30 PM](https://user-images.githubusercontent.com/16461988/129134718-741831f9-ceaa-44a5-b285-42e451062e6d.png)

Reply:
> I have tried something just to understand how this would work - I think we need to decide/add prevention to that.
> 
> Steps:
> 
>     * started a Feature showcase application
> 
>     * waited to get first warning
> 
>     * continued with application - while I was working on it my application was set to EXPIRED
> 
>     * clicked on "Review and summary" and then Submitted
> 
>     * The application has done through
> 
> 
> I'm unsure if we should:
> 
>     * Look the application at this point - disabling the "Review and Submit"
> 
>     * Don't expire application if the section is alive - probably more tricky to do
> 
>     * Allow going until the resume page, but show problem to submit - Application is expired
> 

Good point, although hopefully it wouldn't happen too much in real life since expiries will be weeks, not minutes! 😄

My thought is that it should be allowed for application submissions. (ie. the current behaviour is okay) Maybe I'm just being lazy, but I think if you are in actively working on an application, it shouldn't expire under you.

It's a bit different for renewals, since the license SHOULD expire, regardless of your application state.

It should be possible to set a condition on the actions for templates for which this is a genuine concern (i.e. don't run changeStatus action unless `status != EXPIRED`, something like that). It wouldn't give the user much feedback (actually, what does happen if the submission page returns something other than "SUBMITTED" after the trigger has reset? Not sure) but it would prevent being able to submit after expiry.

Reply:
> it wouldn't happen too much in real life since expiries will be weeks, not minutes!

I think that could happen - imagine someone saves a page for an application and just loads it after it has been expired. Same problem would occur

Reply:
> what does happen if the submission page returns something other than "SUBMITTED" after the trigger has reset? Not sure) but it would prevent being able to submit after expiry.

Reply:
Thanks. 

So if I'm understanding correctly:
- a conditional "changeStatus" action would prevent the status being set to SUBMITTED (if required), so we're fine there
- if this happens the user wouldn't actually know on the submission page, but they'd see when they went to their Dashboard or List (or checked their email)

I think this is fine for now (feel free to make an issue to address it at some point), but I guess we'd just need to add an additional check around here and display something else if status != SUBMITTED:

![Screen Shot 2021-08-12 at 4 21 05 PM](https://user-images.githubusercontent.com/5456533/129137825-b3297bc2-59a7-4e07-ac13-29147181a6e8.png)

(So it's just front-end logic, nothing needed to change in back-end right?)

Reply:
Sure, if changing the changeStatus to check for that it should be enough. But that will need to be done as a core-action
and updated in all templates so we have the same behaviour elsewhere

Reply:
> (So it's just front-end logic, nothing needed to change in back-end right?)

Reply:
> Sure, if changing the changeStatus to check for that it should be enough. But that will need to be done as a core-action
> and updated in all templates so we have the same behaviour elsewhere

No, I don't think that would be core. As I said, I think current behaviour is fine for most applications, it's only for renewals when the "product" is expiring (rather than a stale Draft), that you'd want to be strict about it.

Maybe that's not the case, but l'd rather deal with it as a requirement when we know it's actually needed.